services/api/src/utils/mailer.js

```diff
@@ -1,15 +1,107 @@
 const postmark = require('postmark');
-
 const marked = require('marked');
 const Mustache = require('mustache');
+const { memoize } = require('lodash');
+const fm = require('front-matter');
 const fs = require('fs').promises;
 const path = require('path');
 
 const config = require('@bedrockio/config');
 const { logger } = require('@bedrockio/instrumentation');
 
-const POSTMARK_FROM = config.get('POSTMARK_FROM');
-const ENV_NAME = config.get('ENV_NAME');
+const ENV = config.getAll();
+
+const { ENV_NAME, POSTMARK_FROM, POSTMARK_API_KEY, POSTMARK_SEND_DEV } = ENV;
+
+async function sendTemplatedMail({ template, layout = 'layout.html', to, ...options }) {
+  const textTemplate = template.replace(/\.md$/, '.txt');
+  let [
+    { body: layoutBody, meta: layoutMeta },
+    { body: templateBody, meta: templateMeta },
+    { body: textBody, meta: textMeta },
+  ] = await Promise.all([fetchTemplate(layout), fetchTemplate(template), fetchTemplate(textTemplate)]);
+
+  if (!templateBody && !textBody) {
+    logger.error(`Could not load template ${template}.`);
+    return;
+  }
+
+  options = {
+    ...ENV,
+    ...layoutMeta,
+    ...textMeta,
+    ...templateMeta,
+    ...options,
+  };
+
+  const body = render(templateBody, options);
+
+  const html = render(layoutBody, {
+    ...options,
+    content: body,
+  });
+
+  if (!textBody) {
+    textBody = htmlToText(body);
+  }
+
+  const text = render(textBody, options);
+  const subject = render(options.subject, options);
+
+  if (!to && options.user) {
+    const { user } = options;
+    to = `"${user.name}" <${user.email}>`;
+  }
+
+  await sendMail({
+    to,
+    text,
+    html,
+    body,
+    subject,
+    options,
+  });
+}
+
+async function sendMail({ to, subject, html, text, body, options }) {
+  if (canSendMail()) {
```

Comment:
don't love this construct, wondering if that could be written simpler where you return early.
i love return early, think thats generally the correct pattern.

Reply:
100% disagree :) ... there are some cases where returning up front is more readable but if/else exists for a reason I much prefer it in common cases.

Reply:
besides in this case what would it be like:

```
if (canSendMail()) {
  // do the main thing
}
// do the non-main thing as a fallback? that's almost reversed from what you'd expect
```

readability is subjective but it's also situational

Reply:
lets leave that behind ... as you said `readability is subjective`

Reply:
kk how about this then... inlined everything and instead made `sendEmail` actually do what it claims it does